docs/integrations/query.mdx

```diff
@@ -92,6 +92,71 @@ const UserData = () => {
 }
 ```
 
+## Referencing the same instance of Query Client in your project
+
+Perhaps you have some custom hooks in your project that utilises the `useQueryClient()` hook to obtain the `QueryClient` object and call its methods.
+
+To ensure that you reference the same `QueryClient` object, be sure to wrap the root of your project in a `<Provider>` and initialise `queryClientAtom` with the same `queryClient` value you provided to `QueryClientProvider`.
+
+Without this step, `useQueryAtom` will reference a seperate `QueryClient` from any hooks that utilise the `useQueryClient()` hook to get the queryClient.
+
+### Example
+
+In the example below, we have a mutation hook, `useTodoMutation` and a query `todosAtom`.
+
+We included an initialisation step in our root `<App>` node.
+
+Although they reference methods same query key (`'todos'`), the `onSuccess` invalidation in `useTodoMutation` will not trigger **if the `Provider` initialisation step was not done.**
+
+This will result in `todosAtom` showing stale data as it was not prompted to refetch.
+
+```jsx
+import {
+  useMutation,
+  useQueryClient,
+  QueryClient,
+  QueryClientProvider,
+} from 'react-query'
+import { atomWithQuery } from 'jotai/query'
+
+const queryClient = new QueryClient()
+export const App = () => {
+  return (
+    <QueryClientProvider client={queryClient}>
+      {/* This Provider initalization step is needed so that we reference the same
+      queryClient in both atomWithQuery and other parts of the app. Without this,
+      our useQueryClient() hook will return a different QueryClient object */}
+      <Provider initialValues={[[queryClientAtom, queryClient]]}>
+        <App />
+      </Provider>
+    </QueryClientProvider>
+  )
+}
+
+export const todosAtom = atomWithQuery((get) => {
+  return {
+    queryKey: ['todos'],
+    queryFn: () => fetch('/todos'),
+  }
+})
+
+export const useTodoMutation = () => {
+  const queryClient = useQueryClient()
+
+  return useMutation(
```

Comment:
`useMutation` is unavoidable at this point.
But, from the `jotai/query` perspective (which is not tied to React),
we would like to have `atomWithMutation`. (We have it in `jotai/urql`.)
Would you be interested in contributing in it?? Issue 👉 #309

Reply:
Thanks, would love to attempt and contribute to it. 

Will try to implement a wrapper in my existing project and make a new PR when its ready! 

Thanks for the quick review, great library! 🥳